bentoml/server/api_server.py

```diff
@@ -299,22 +275,17 @@ def bento_service_api_func_wrapper(self, api):
         service_version = self.bento_service.version
 
         def api_func():
-            # Log image files in request if there is any
-            image_paths = self.log_image(request, request_id)
-
-            # _request_to_json parses request as JSON; in case errors, it raises
-            # a 400 exception. (consider 4xx before 5xx.)
-            request_for_log = _request_to_json(request)
```

Comment:
is the plan to move prediction logging out of here and make it work with `Input` directly?

Reply:
Yeah, that is the idea since there was specialized logging for both JSON and image inputs, which made more sense to have in their respective input adapters. I think we'll need to come up with some sort of interface to let input adapters cleanly document their inputs.

Reply:
The problem with that is now the input adapter does not have access to the output of the API request. Usually, the user would expect one prediction log item to have both the inference input and output. We probably want to move the logging logic to `InferenceAPI` and allow adapters to define how to turn a `InferenceTask` and `InferenceResult` into a log record.

I think it might be easier to leave this change on prediction logging for later PR to reduce the scope of this pull request. The sooner we merge the refactoring in this PR, the easier would be for @bojiang to build the adapter internal refactoring base upon this.